include/gz/rendering/base/BaseStorage.hh

```diff
@@ -854,7 +857,12 @@ namespace gz
     typename BaseStore<T, U>::ConstUIter
     BaseStore<T, U>::ConstIterByName(const std::string &_name) const
     {
-      return this->store.find(_name);
+      auto idx = this->storeMap.find(_name);
+      if (idx == this->storeMap.end())
+      {
+        return this->store.end();
+      }
+      return ConstIterByIndex(idx->second);
```

Comment:
I see we're returning iterators. If the iterators are being stored somewhere, we need to be careful since insertion to the `std::vector` might invalidate it. This was not the case for `std::map`, so we might be introducing subtle bugs here.

Reply:
Since the original implementation was based on a map, it makes sense to keep iters and pass them around. Now that we're changing to vector, I think we don't need to be doing this as much. The only place that use these `IterByName` functions internally is `ContainsName`, which can by re-implemented easily.

So how about we deprecate `ConstIterByName` / `IterByName` functions, update doxygen comment for these 2 APIs to warn users not to keep a handle to the iterator as it may change, and add a note to migration guide?

Reply:
ok I created a PR: https://github.com/gazebosim/gz-rendering/pull/832

I didn't deprecate the functions. Instead I added doc with a note that iterators can change / become invalid. I also updated migration guide.